components/public-api/gitpod/v1/envvar.proto

```diff
@@ -0,0 +1,88 @@
+syntax = "proto3";
+
+package gitpod.v1;
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/v1";
+
+service EnvironmentVariableService {
+  // ListEnvironmentVariables returns all environment variables for the
+  // authenticated user.
+  //
+  // +return NOT_FOUND User does not have access to a workspace with the given
+  // ID +return NOT_FOUND Workspace does not exist
+  rpc ListEnvironmentVariables(ListEnvironmentVariablesRequest)
+      returns (ListEnvironmentVariablesResponse) {}
+
+  // UpdateEnvironmentVariable updates an environment variable for the
+  // authenticated user.
+  rpc UpdateEnvironmentVariable(UpdateEnvironmentVariableRequest)
+      returns (UpdateEnvironmentVariableResponse) {}
+
+  // CreateEnvironmentVariable creates a new environment variable for the
+  // authenticated user.
+  rpc CreateEnvironmentVariable(CreateEnvironmentVariableRequest)
+      returns (CreateEnvironmentVariableResponse) {}
+
+  // DeleteEnvironmentVariable deletes an environment variable for the
+  // authenticated user.
+  rpc DeleteEnvironmentVariable(DeleteEnvironmentVariableRequest)
+      returns (DeleteEnvironmentVariableResponse) {}
+}
+
+message UserScope {
```

Comment:
it is global type, i thin we should prefxi it then EnvrionmentVariableUserScope

or put them as nested, i.e.

```
message CreateEnvironmentVariableRequest {
  message UserScope {
...
```

Reply:
given that they are not reusable for Update, i think it should be nested in CreateEnvironmentVariableRequest

Reply:
they are both used in update and create

Reply:
it will incorrect because Update should be partial, please see https://github.com/gitpod-io/gitpod/pull/19067#discussion_r1393986454 and https://www.notion.so/gitpod/Public-API-guideline-a010fa3d494544b78f28945e53f730ac?pvs=4#fe62cdc913df42babfbdc61b46044ed6

Reply:
we might not need and want to allow to update the scope of an env var